frontend/javascripts/oxalis/model/reducers/skeletontracing_reducer_helpers.js

```diff
@@ -524,14 +524,18 @@ export function addTreesAndGroups(
 
     if (allowUpdate) {
       // Assign a new tree name for trees without a name
-      for (const treeId of Object.keys(trees)) {
-        const tree = trees[Number(treeId)];
+      const treeIds = Object.keys(trees).map(treeId => Number(treeId));
+      for (const treeId of treeIds) {
+        const tree = trees[treeId];
         if (tree.name === "") {
           tree.name = generateTreeName(state, tree.timestamp, tree.treeId);
         }
       }
 
-      const needsReassignedIds = Object.keys(skeletonTracing.trees).length > 0;
+      // TreeIds > 1024^2 break webKnossos, see https://github.com/scalableminds/webknossos/issues/5009
+      const hasTreeIdsLargerThanMaximum = treeIds.filter(treeId => treeId > 1048576);
```

Comment:
```suggestion
      const hasTreeIdsLargerThanMaximum = treeIds.some(treeId => treeId > 1048576);
```

Otherwise, this would return an array, which is always truthy (so, the check in the next line would always evaluate to true). Additionally, this should be better perf-wise (since this can do an early-out).

Reply:
:man_facepalming: I forgot to add `.length` in the end. Thanks for catching that and your suggestion is even better with the early-out :+1: